opencti-platform/opencti-graphql/src/manager/ingestionManager.ts

```diff
@@ -299,37 +329,41 @@ const csvDataToObjects = async (csvBuffer: Buffer | string, ingestion: BasicStor
+    const csvMapperParsed = parseCsvMapper(csvMapper);
+    csvMapperParsed.user_chosen_markings = ingestion.markings ?? [];
 
-  const { data, addedLast } = await fetchCsvFromUrl(csvMapperParsed, ingestion);
-  const isUnchangedData = bcrypt.compareSync(data.toString(), ingestion.current_state_hash ?? '');
-  if (isUnchangedData) {
-    return;
-  }
+    const { data, addedLast } = await fetchCsvFromUrl(csvMapperParsed, ingestion);
+    const isUnchangedData = bcrypt.compareSync(data.toString(), ingestion.current_state_hash ?? '');
+    if (isUnchangedData) {
+      return;
+    }
 
-  const objects = await csvDataToObjects(data, ingestion, csvMapperParsed, context);
-  const bundleSize = 1000;
-  for (let index = 0; index < objects.length; index += bundleSize) {
+    const objects = await csvDataToObjects(data, ingestion, csvMapperParsed, context);
+    const bundleSize = 1000;
+    for (let index = 0; index < objects.length; index += bundleSize) {
     // Filter objects already added to queue
-    const splitBundle = objects.slice(index, index + bundleSize);
-    const bundle = { type: 'bundle', id: `bundle--${uuidv4()}`, objects: splitBundle };
-    const stixBundle = JSON.stringify(bundle);
-    const content = Buffer.from(stixBundle, 'utf-8').toString('base64');
-    const friendlyName = 'CSV feed Ingestion';
-    const work = await createWork(context, user, IMPORT_CSV_CONNECTOR, friendlyName, IMPORT_CSV_CONNECTOR.id) as unknown as Work;
-    await updateExpectationsNumber(context, user, work.id, 1);
-    await pushToWorkerForSync({ type: 'bundle', applicant_id: ingestion.user_id ?? OPENCTI_SYSTEM_UUID, work_id: work.id, content, update: true });
-  }
+      const splitBundle = objects.slice(index, index + bundleSize);
+      const bundle = { type: 'bundle', id: `bundle--${uuidv4()}`, objects: splitBundle };
+      const stixBundle = JSON.stringify(bundle);
+      const content = Buffer.from(stixBundle, 'utf-8').toString('base64');
+      const friendlyName = 'CSV feed Ingestion';
+      const work = await createWork(context, user, IMPORT_CSV_CONNECTOR, friendlyName, IMPORT_CSV_CONNECTOR.id) as unknown as Work;
+      await updateExpectationsNumber(context, user, work.id, 1);
+      await pushToWorkerForSync({ type: 'bundle', applicant_id: ingestion.user_id ?? OPENCTI_SYSTEM_UUID, work_id: work.id, content, update: true });
+    }
 
-  // Update the state
-  const hashedIncomingData = bcrypt.hashSync(data.toString());
-  await patchCsvIngestion(context, SYSTEM_USER, ingestion.internal_id, {
-    current_state_hash: hashedIncomingData,
-    added_after_start: utcDate(addedLast)
-  });
+    // Update the state
+    const hashedIncomingData = bcrypt.hashSync(data.toString());
+    await patchCsvIngestion(context, SYSTEM_USER, ingestion.internal_id, {
+      current_state_hash: hashedIncomingData,
+      added_after_start: utcDate(addedLast),
+    });
+  } catch (e) {
```

Comment:
why do we need a try catch here ? I can see a catch in CsvExecutor. Same question for RssDataHandler

Reply:
no needed, you're right